src/push_service/profile.rs

```diff
@@ -88,18 +89,17 @@ struct SignalServiceProfileWrite<'s> {
 impl PushService {
     pub async fn retrieve_profile_by_id(
         &mut self,
-        address: ServiceAddress,
+        address: Aci,
         profile_key: Option<zkgroup::profiles::ProfileKey>,
     ) -> Result<SignalServiceProfile, ServiceError> {
         let endpoint = if let Some(key) = profile_key {
-            let version = bincode::serialize(&key.get_profile_key_version(
-                address.aci().expect("profile by ACI ProtocolAddress"),
-            ))?;
+            let version =
+                bincode::serialize(&key.get_profile_key_version(address))?;
             let version = std::str::from_utf8(&version)
                 .expect("hex encoded profile key version");
-            format!("/v1/profile/{}/{}", address.uuid, version)
+            format!("/v1/profile/{}/{}", address.service_id_string(), version)
         } else {
-            format!("/v1/profile/{}", address.uuid)
+            format!("/v1/profile/{}", address.service_id_string())
```

Comment:
Here we *do* have the domain separation prefix; is this correct?

Reply:
IIUC, profiles are ACI-only (when someone contacts you by phone number, it's from-ACI-to-PNI message) so this looks correct to me.

What should be checked is how "I want to send a message to a phone number" is handled; IIRC there's no profile information fetched before at least sending the message...

Reply:
Then to avoid the back-and-forth conversion, we can switch to `raw_uuid()`.